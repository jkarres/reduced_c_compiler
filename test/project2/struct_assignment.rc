// test struct assignment and initialization

structdef S {
  int i;
  float f;
  bool[5] ba;

  function : void print() {
    cout << "<S";
    cout << " i:" << this.i;
    cout << " f:" << this.f;
    cout << " ba:<";
    int i;
    for (i = 0 ;  ; ++i) {
      cout << this.ba[i];
      if (i == 4) { break; }
      cout << " ";
    }
    cout << ">>" << endl;
  }

};

S[2] sa;

function : void main() {
  cout << "sa[0]:";
  sa[0].print();

  cout << "s[1]:";
  sa[1].print();

  sa[0].i = 11;
  sa[0].f = 3293.92;
  sa[0].ba[0] = true;
  sa[0].ba[1] = true;
  sa[0].ba[2] = false;
  sa[0].ba[3] = true;
  sa[0].ba[4] = true;
  
  cout << "sa[0]:";
  sa[0].print();

  cout << "sa[1] = sa[0];" << endl;
  sa[1] = sa[0];


  cout << "sa[1]:";
  sa[1].print();

  cout << "Creating ls on the stack, initialized from sa[1]." << endl;
  S ls = sa[1];
  cout << "ls:";
  ls.print();

}

