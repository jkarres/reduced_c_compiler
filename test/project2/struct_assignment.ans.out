sa[0]:<S i:0 f:0.00 ba:<false false false false false>>
s[1]:<S i:0 f:0.00 ba:<false false false false false>>
sa[0]:<S i:11 f:3293.92 ba:<true true false true true>>
sa[1] = sa[0];
sa[1]:<S i:11 f:3293.92 ba:<true true false true true>>
Creating ls on the stack, initialized from sa[1].
ls:<S i:11 f:3293.92 ba:<true true false true true>>
